src/main/java/org/scoverage/plugin/SCoverageCheckMojo.java

```diff
@@ -168,31 +223,27 @@ public void execute() throws MojoFailureException
         int invokedBranchesCount = coverage.invokedBranchesCount();
         int invokedStatementCount = coverage.invokedStatementCount();
 
-        getLog().info( String.format( "Statement coverage.: %s%%", coverage.statementCoverageFormatted() ) );
-        getLog().info( String.format( "Branch coverage....: %s%%", coverage.branchCoverageFormatted() ) );
         getLog().debug( String.format( "invokedBranchesCount:%d / branchCount:%d, invokedStatementCount:%d / statementCount:%d",
                                       invokedBranchesCount, branchCount, invokedStatementCount, statementCount ) );
-        if ( minimumCoverage > 0.0 )
+
+        boolean ok = checkCoverage( getLog(), "Total", coverage,
+                                    minimumCoverage, minimumCoverageBranchTotal, true );
+        for ( MeasuredPackage pkgCoverage : CollectionConverters.asJava( coverage.packages() ) )
+        {
+            if ( !checkCoverage( getLog(), "Package:" + pkgCoverage.name(), pkgCoverage,
+                                 minimumCoverageStmtPerPackage, minimumCoverageBranchPerPackage, false ) )
+                ok = false;
```

Comment:
seems like this will continue even after after first `false`
should we instead use something like
 `CollectionConverters.asJava( coverage.packages() ).stream().allMatch`?
or do you want it to run for each package no matter what?
same question for files

Reply:
good question. we can either report until first failure (in which case the user would fix something and then find out there's more), or report everything. options:
- report everything
- report only first failure
- put the choice under yet another parameter.

Reply:
I personally generate a report on a failure, to see what's not covered. So the first failure seems fine to me. 
But I can also see a point of having all the packages/files checked at once.

How about we do fail-fast in this PR and then we can do a separate PR to add an extra flag, like `failFast` with `true` by default?

Reply:
@jozic good morning and happy new year.

i finally got around to working on this, and here's what i realized: we will have reporting on all packages and files if it's successful but will stop at the first error if it happens (with a single [ERROR] message after possibly a bunch of [INFO]/[DEBUG]). which is a bit weird since the opposite is probably of more interest.

so, i see several alternatives:
- report everything, just like now
- report everything, except use [DEBUG] for second and subsequent errors
- collect all info/debug messages until the first error; if there are no errors, only then log the info/debug.

thoughts?

Reply:
Thanks @kitbellew, happy New Year to you as well!

Good point, i guess reporting packages and files as debug for success and error for failure is a fair game. So we should probably leave it as you have it )

Reply:
@jozic all done and pushed.